freqtrade/persistence/models.py

```diff
@@ -254,11 +257,20 @@ class LocalTrade():
     max_rate: float = 0.0
     # Lowest price reached
     min_rate: float = 0.0
-    sell_reason: str = ''
-    sell_order_status: str = ''
+    close_reason: str = ''   
+    close_order_status: str = '' 
     strategy: str = ''
     timeframe: Optional[int] = None
 
+    #Margin trading properties
+    leverage: Optional[float] = None
+    borrowed: float = 0
+    borrowed_currency: float = None
```

Comment:
I THINK you'll need collateral currency as well (or instead of this?).

AFAIK, you can Long/Short  ETH/BTC - while using USDT as collateral - so your USDT balance will determine the liquidation price - while it's not directly involved in your trade.

Reply:
I added the field `collateral_currency` for now, and changed both these currencies to string

Reply:
> AFAIK, you can Long/Short ETH/BTC - while using USDT as collateral - so your USDT balance will determine the liquidation price - while it's not directly involved in your trade.

If were going to allow those then we would have to allow freqtrade to work with multiple base currencies (because if your base is USDT and you short ETH on ETH/BTC, you'll either have to buy back extra ETH, or leave a little bit of BTC), or we'll have to immediately sell the remaining ETH or BTC, which the only use I can think of for that is if the exchange has run out of USDT to lend out for the pair your trading, so then you would need to borrow another currency like BTC if you wanted to buy ETH. That could be a nice feature, but it would be a decent amount of extra work, and I feel like the logic would be kinda different anyway.

Reply:
> AFAIK, you can Long/Short ETH/BTC - while using USDT as collateral - so your USDT balance will determine the liquidation price - while it's not directly involved in your trade.

If were going to allow those then we would have to allow freqtrade to work with multiple base currencies (because if your base is USDT and you short ETH on ETH/BTC, you'll either have to buy back extra ETH, or leave a little bit of BTC), or we'll have to immediately sell the remaining ETH or BTC, which the only use I can think of for that is if the exchange has run out of USDT to lend out for the pair your trading, so then you would need to borrow another currency like BTC if you wanted to buy ETH. That could be a nice feature, but it would be a decent amount of extra work, and I feel like the logic would be kinda different anyway.

Reply:
won't you "repay" the moment you sell your trade (or it get's liquidated) ? 

but the way you're describing it youre probably right - it's not worth the extra effort for the moment

Reply:
> In Cross Margin Mode, the margin can only be shared between the same type of asset. For example, in Cross Margin Mode, all BUSD in the USDⓈ-M Futures Wallet can be used for all BUSD-margined contracts.

[source](https://www.binance.com/en/support/faq/360033162192)